src/history.rs

```diff
@@ -196,6 +196,18 @@ pub fn rewrite_commit(
         parents,
     )?;
 
+    if let Ok((sig, _)) = repo.extract_signature(&base.id(), None) {
```

Comment:
`None` here and in the `commit_signed` below refers to the name of the signature field. Looks like that can be customized. Roundtrips will still not work if there are commits that use a non-standard name for that field. I don't know how to get git to tell us the name of that field... really ideally we'd have an API to list all header fields, and preserve them all while only adjusting the `tree`, but I am not sure if that is possible.

Reply:
That's quite interesting: We could maybe at some point even use that for out own metadata. Good to know.

About getting all extra header fields: Worst case we have to parse the commit buffer ourselves. But I think this is good enough for now.

One little thing: The constant `CACHE_VERSION` in `src/cache.rs` should be incremented whenever we make a change that can affect the output sha's of any filter. This is to avoid strange effects that could happen when somebody uses a cache that was computed using the old version and things could get mixed up.